src/geom/face_quad4.C

```diff
@@ -265,6 +265,57 @@ void Quad4::connectivity(const unsigned int libmesh_dbg_var(sf),
 
 
 
+Point Quad4::true_centroid () const
+{
+  // Note: this function does a substantially similar behavior to the
+  // volume() calculation, so it would be good if they could share
+  // code somehow without losing any performance by introducing extra
+  // function calls, etc.
```

Comment:
Inline function in an anonymous namespace?

Reply:
Yes that's probably the way to go, but my thinking has shifted a bit since writing this comment, since only a few lines of code that are unlikely to change actually need to be duplicated. 

What I think will actually be more useful is a stand-alone `compute_quad4_centroid(Point& p0, Point& p1, Point& p2, Point& p3);` function that would be called by Quad4, but also by 3D Elems with Quad faces (Pyramid, Hex, prism) since the formulas for the 3D centroids depend explicitly on the centroids of the faces. But that's an optimization I'd like to save for a later PR.